torsiondrive/qm_engine.py

```diff
@@ -144,6 +145,44 @@ def optimize_geomeTRIC(self):
     def load_native_output(self):
         return Molecule()
 
+class EngineOpenMM(QMEngine):
+    def load_input(self, input_file):
+        """Input file is the name of the pdb file with the coords in we also require that the xml has the same name"""
+
+        self.m_pdb = Molecule(input_file)[0]
+        self.M = copy.deepcopy(self.m_pdb)
+
+        xml_name = os.path.splitext(input_file)[0] + '.xml'
+        # Check the xml file is present
+        assert os.path.exists(xml_name) is True, "OpenMM requires a pdb and xml file, ensure you have both in the current folder with the same prefix"
+        with open(xml_name) as f:
+            self.xml = f.read()
+
+    def write_input(self):
+        """write the pdb and xml file for geometric assuming the rename is UNK"""
+
+        self.m_pdb.xyzs[0] = self.M.xyzs[0]
+        self.m_pdb.write('tdrive.pdb')
+        with open('tdrive.xml', 'w') as out:
+            for line in self.xml:
+                out.write(line)
+
+    def optimize_geomeTRIC(self):
+        """ run the constrained optimization using geomeTRIC package, in 3 steps:
+        1. Write a constraints.txt file.
+        2. Write a gradient job input file.
+        3. Run the job
+        """
+        # sep 1
+        self.write_constraints_txt()
+        # step 2
+        self.write_input()
+        # set3
+        self.run('geometric-optimize --prefix tdrive --qccnv --reset --epsilon 0.0 --enforce 0.1 --qdata --pdb '
+                 'tdrive.pdb --openmm tdrive.xml constraints.txt',
```

Comment:
I found a hidden bug here when using work_queue.
When the input files are named the same as the prefix, which is `tdrive` here, geomeTRIC to automatically rename the output xyz file to `tdrive_optim.xyz`, but the work_queue is expecting the file named `tdrive.xyz`, which will cause the tasks to fail.
Please merge the branch `pr` to fix this.